docs/2024/puzzles/day08.md

```diff
@@ -6,6 +6,129 @@ import Solver from "../../../../../website/src/components/Solver.js"
 
 https://adventofcode.com/2024/day/8
 
+## Solution summary
+
+1. Parse the map to identify the locations of all the antennae,
+grouped by their frequency.
+2. Find all the antinode locations within the map boundary.
+3. Count the number of distinct antinodes.
+
+### Data model
+
+First, we'll define some case classes to represent the antenna map.
+We could use tuples, but the named classes help readability.
+
+#### Antenna map
+
+An antenna map contains the width and height of the map, along with
+a sequence of sequences of antenna locations. Each sequence of antenna
+locations corresponds to a particular frequency (i.e. a particular
+letter or digit on the map.)
+
+```scala 3
+final case class AntennaMap(
```

Comment:
I would suggest removing `final` here and on your other case classes — unless you feel strongly about including it. I know there are work codebases where people are adamant about declaring case classes `final`, but it seems like noise me in an AoC context.

If you do feel strongly about including it, I'd suggest adding some brief wording to justify it.

Reply:
force of habit, removed.